### PR TITLE
feat: implement `Channel` object

### PR DESCRIPTION
**Pull request by Dworv**
This PR adds the `Channel` object, as well as all of its required dataclasses such as but not limited to `Overwrite` and `ChannelFlags`. It also adds some needed message objects like `MessageType` and `MessageFlags` among other things.

### CODE REVIEW

File: retux/client/resources/channel.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+
+    Stops counting at 50.
+    """
+    thread_metadata: dict | ThreadMetadata | None = field(
+        default=None, converter=ThreadMetadata
+    )
+    """Thread-specific fields not needed by other channels."""
+    member: ThreadMember | None = field(default=None)
+    """
+    Thread member object for the current user if they have joined the thread
```
Comment (i0bs) [inline retux/client/resources/channel.py:345]:
```suggestion
    The thread member representation of the user if they have joined the thread.
```

File: retux/client/resources/channel.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+    """
+    default_auto_archive_duration: int | None = field(default=None)
+    """
+    Default archive duration for clients in minutes.
+    
+    Can be set to 60, 1440, 4320, 10080.
+    """
+    permissions: str | None = field(default=None)
+    """
+    Computed permissions for the invoking user in the channel, including overwrites.
```
Comment (i0bs) [inline retux/client/resources/channel.py:357]:
```suggestion
    The computed permissions for the invoking user in the channel, including any overwrites.
```

**Review by i0bs — CHANGES_REQUESTED** [review]
Please note some reviews target docstrings on attrs, please make sure they're carried over to the class docstring.

**Review by i0bs — CHANGES_REQUESTED** [review]
More clarity on `attrs` usage:

I found out that `converter` is not required when you're dealing with lists or dictionaries, as `attrs` will automatically try to work with these. It is only on custom-defined classes where you need this. I've reflected these changes to the main branch already for the other classes, so you don't have to worry about those.

*Additionally,* `attrs` will apparently complain if you try to convert a given `None` value, so any and every field with a default conversion needs to be checked. for any `flags` fields, my recommendation is setting them to `0` as the default.

File: retux/client/resources/channel.py [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+    """
+
+    channel_id: str | Snowflake = field(converter=Snowflake)
+    """The ID of the source channel."""
+    webhook_id: str | Snowflake = field(converter=Snowflake)
+    """The ID of the created target webhook."""
+
+
+@define()
+class Channel:
```
Comment (i0bs) [inline retux/client/resources/channel.py:181]:
```suggestion
class Channel(Object):
```

File: retux/client/resources/__init__.py [hunk 4]
```diff
@@ -3,4 +3,5 @@
 from .components import *  # noqa
 
 # Resource modules to be added when all dependencies are added as well:
-# from .application
+# from .application import * # noqa
+# from .channel import * # noqa
```
Comment (EdVraz) [inline retux/client/resources/__init__.py:7]:
```suggestion
# from .channel import * # noqa

```
Follow PEP

File: retux/client/resources/channel.py [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+    """The channel's voice region ID if present, set to automatic when left as `None`."""
+    video_quality_mode: int | VideoQualityMode | None = field(
+        converter=VideoQualityMode, default=None
+    )
+    """The video quality mode of the voice channel."""
+    message_count: int | None = field(default=None)
+    """"
+    The approximated amount of messages in a thread.
+
+    Stops counting at 50.
```
Comment (i0bs) [inline retux/client/resources/channel.py:477]:
```suggestion
    Stops counting at `50`.
```

File: retux/client/resources/channel.py [hunk 6]
```diff
@@ -0,0 +1,508 @@
...
+    """
+    member_count: int | None = field(default=None)
+    """
+    The approximated amount of users in a thread.
+
+    Stops counting at 50.
```
Comment (i0bs) [inline retux/client/resources/channel.py:483]:
```suggestion
    Stops counting at `50`.
```

**Review by i0bs — CHANGES_REQUESTED** [review]

**Review by i0bs — COMMENTED** [review]
Please remove the `=` documentation on `VideoQualityMode` and `ChannelType`. I will wait for this Pull Request to be flagged as ready for review before taking a final look.

**Review by i0bs — CHANGES_REQUESTED** [review]

**Review by i0bs — APPROVED** [review]
LGTM, thank you for contributing to retux!

**Pull request merged**